docs/design/service_traffic_policy.md

```diff
@@ -315,3 +297,128 @@ where packet leaves the node and goes back to the external entity that initiated
 
 ## Sources
 - https://www.asykim.com/blog/deep-dive-into-kubernetes-external-traffic-policies
+
+## Internal Traffic Policy
+
+Service Internal Traffic Policy (ITP) is a feature that can be enabled on a kubernetes service type object. This feature imposes restrictions on traffic that originates internally by routing it only to endpoints that are local to the node from where the traffic originated from. Here "internal" traffic means traffic originating from pods and nodes within the cluster. See https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy/ for more details.
+
+```
+NOTE1: ITP is applicable only to service of type "ClusterIP" meaning it has no effect on services of type NodePorts, ExternalIPs or LoadBalancers.
+
+NOTE2: Unlike ETP, it is not necessary that the srcIP be preserved in case of ITP.
+```
+
+By default ITP is of type `Cluster` on all services; meaning internal traffic will be routed to any of the endpoints for that service. When it is set to `Local`, only local endpoints are considered. The way this is implemented in OVN-K is by filtering out endpoints from the load balancer on the node switches for `ClusterIP` services .
+
+## InternalTrafficPolicy=Local
+
+This feature is implemented exactly in the same way for both the gateway modes.
+
+### Pod Traffic
+
+This section will cover the networking entities hit when traffic travels from an OVN pod via a clusterIP service to either host networked pods or cluster networked pods.
+
+We have a service of type `NodePort` where `internalTrafficPolicy: Local`:
+
```

Comment:
hmm itp is only relevant to svc of type clusterIP so why adding example with np ?

Reply:
right for two reasons, firstly a nodePort service is also a clusterIP service. Secondly we can see that the endpoints for nodeports are not filtered on the switches which is an important point to remember. Using a simple clusterIP service type is not that complicated so I thought I'd show the complicated use case which covers the simple one as well.